aten/src/ATen/core/DistributionsHelper.h

```diff
@@ -126,6 +126,65 @@ struct uniform_real_distribution {
     T to_;
 };
 
+#define DISTRIBUTION_HELPER_GENERATE_HAS_MEMBER(member)              \
+template <typename T>                                                \
```

Comment:
This can be more easily implemented with `guts::void_t`, something like

```cpp
template<class T, class Enable = void> struct has_member_X : std::false_type {};
template<class T> struct has_member_X<T, guts::void_t<T::X>> : std::true_type {};
```